src/descriptors/element_clip_edge.js

```diff
@@ -0,0 +1,195 @@
+	ensure.signature(arguments, []);
+
+	var rect = this._element.getRawStyle("clip");
+
+	if (rect === "") {
+		// As a fallback for IE8 for when it can't fork over the original clip css style, try generating a clip rect
+		// using clip components that currentStyle may have.  If we see non-empty strings for all four components, we'll
+		// build out a clip rect expression here ...
+
+		var clipLeft = this._element.getRawStyle("clip-left");
+		var clipRight = this._element.getRawStyle("clip-right");
+		var clipBottom = this._element.getRawStyle("clip-bottom");
+		var clipTop = this._element.getRawStyle("clip-top");
+
+		if (clipLeft && clipRight && clipBottom && clipTop) {
+			rect = "rect(" + clipTop + " " + clipRight + " " + clipBottom + " " + clipLeft + ")";
+		}
+	}
+
+	for (var i = 0, ii = UNSET_CLIP_STYLES.length; i < ii; i++) {
+		if (UNSET_CLIP_STYLES[i] === rect) {
+			return null;
+		}
+	}
+
+	var matches = rect.match(CLIP_RECT_PATTERN);
+	if (!matches) {
+		ensure.unreachable("Unknown clip css style: " + rect);
+	}
+
+	// values in a clip's rect may be a css length or "auto" which means "clip over the edge's border"
+	var clipTopPx = this.computeClipTopPxHeight(matches[1]);
+	var clipRightPx = this.computeClipRightPxWidth(matches[2]);
+	var clipBottomPx = this.computeClipBottomPxHeight(matches[3]);
+	var clipLeftPx = this.computeClipLeftPxWidth(matches[4]);
+
+	var domElement = this._element.toDomElement();
+	var boundingRect = domElement.getBoundingClientRect();
+
+	var clipRect = {
+		left: boundingRect.left + clipLeftPx,
+		right: boundingRect.left + clipRightPx,
+
+		top: boundingRect.top + clipTopPx,
+		bottom: boundingRect.top + clipBottomPx
+	};
+
+	clipRect.width = clipRect.right - clipRect.left;
+	clipRect.height = clipRect.bottom - clipRect.top;
+
+	return clipRect;
+};
+
+var LENGTH_EXPR_PATTERN = /([\-\+]?[0-9\.]+)([a-zA-Z]+)/;
+
+ElementClipEdge.prototype.computeCssPxForLengthInElement = function computeCssPxForLengthInElement(lengthExpr) {
+	var matches = lengthExpr.match(LENGTH_EXPR_PATTERN);
+
+	if(!matches) {
+		ensure.unreachable("CSS length expression expected, got " + lengthExpr);
+	}
+
+	// convert the parsed number part of the lengthExpr to px by multiplying it by the computed ratio of lengthExpr's
+	// css unit to css px units.
+	return parseFloat(matches[1]) * this.computeCssUnitToCssPxRatio(matches[2]);
+};
+
+ElementClipEdge.prototype.computeClipTopPxHeight = function computeClipTopPxHeight(lengthExpr) {
+	return this.computeCssPxForLengthInElement((lengthExpr === "auto") ? "0px" : lengthExpr);
+};
+
+ElementClipEdge.prototype.computeClipRightPxWidth = function computeClipRightPxWidth(lengthExpr) {
+	if(lengthExpr === "auto") {
+		// "auto" for clip rect's right component will be the width of the element, enclosing the borders but not the
+		// margins.  offsetWidth gives us this value
+
+		return this._element.toDomElement().offsetWidth;
```

Comment:
You can take advantage of existing descriptors here: `return this._element.width.value();`